python_ta/checkers/infinite_loop_checker.py

```diff
@@ -1,13 +1,24 @@
 """Check for infinite while loops."""
 
 import itertools
-from typing import Optional
+from typing import Optional, Union
 
 from astroid import BoundMethod, InferenceError, UnboundMethod, bases, nodes, util
 from pylint.checkers import BaseChecker, utils
 from pylint.interfaces import INFERENCE
 from pylint.lint import PyLinter
 
+immutable_types = (
```

Comment:
use the `ALL_CAPS` naming style for this, since it's a constant. Also, you can make this a set